flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/LoopFlowUtil.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.flowbased_computation.impl;
+
+import com.farao_community.farao.balances_adjustment.util.CountryArea;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_file.CracFile;
+import com.farao_community.farao.data.flowbased_domain.DataMonitoredBranch;
+import com.farao_community.farao.data.flowbased_domain.DataPtdfPerCountry;
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationParameters;
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationProvider;
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationResult;
+import com.farao_community.farao.flowbased_computation.glsk_provider.GlskProvider;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Network;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Loop flow util is used
+ * - for loop flow calculation (used in search tree rao for CCR Core), and
+ * - for Min RAM adjustment calculation
+ * @author Pengbo Wang {@literal <pengbo.wang at rte-international.com>}
+ */
+public final class LoopFlowUtil {
```

Comment:
`todo : move in flowbased-computation-api, once the api is based only on Crac and not on CracFile`